functions/go/set-project-id/main.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+
+	kptv1 "github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/fix/v1"
+)
+
+var (
+	kccAPIVersionRegex = regexp.MustCompile(`^([\w]+)\.cnrm\.cloud\.google\.com\/[\w]+$`)
+)
+
+const (
+	projectIDSetterName = "project-id"
+	projectIDAnnotation = "cnrm.cloud.google.com/project-id"
+)
+
+func findSetterNode(nodes []*yaml.RNode, path string) (*yaml.RNode, error) {
+	for _, node := range nodes {
+		np := node.GetAnnotations()[kioutil.PathAnnotation]
+		if np == path {
+			return node, nil
+		}
+	}
+	return nil, fmt.Errorf(`file %s doesn't exist, please ensure the file specified in "configPath" exists and retry`, path)
+}
+
+func findKptfile(nodes []*yaml.RNode) (*kptv1.KptFile, error) {
+	for _, node := range nodes {
+		if node.GetAnnotations()[kioutil.PathAnnotation] == kptv1.KptFileName {
+			kf, err := kptv1.ReadFile(node)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read Kptfile: %v", err)
```

Comment:
Please use `%w` and below.
See R1 in the [style guide](https://github.com/GoogleContainerTools/kpt/blob/main/docs/style-guides/errors.md)

Reply:
Done.